### ISSUE TITLE
<ctrl-c> not stopping text paste

### ISSUE THREAD

**Issue by Ashutosh-Tripathy** [neovim/neovim#13031]
- `nvim --version`:
NVIM v0.4.4
Build type: Release
LuaJIT 2.0.5
.....
Compiled by travis@Traviss-Mac-6.local

Features: +acl +iconv +tui
- `vim -u DEFAULTS` (version: ) behaves differently?
  No
- Operating system/version:
  macOS Catalina 10.15.7
- Terminal name/version:
  Terminal 2.10
- `$TERM`:
  xterm-256color

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC

1) Copy text from big log file (15k+ lines) in system clipboard ("+Y)
2) I want to search a text (using "/") and by mistake paste from system clipboard. 
Now It will take few minutes to complete the paste and I am not able to stop it using <c-c> or <c-d> 
(I  also tried pressing these keys multiple times).
 I can see same behaviour (not able to stop paste operation) when pasting large text from system clipboard in insert mode. 
```
### Actual behaviour
I am unable to stop text pasting. Killing neovim or closing window is only option left.
### Expected behaviour
I should be able to stop operation using ctrl-c.

**Comment by zeertzjq** [comment]
What keys did you use to paste from system clipboard?

**Comment by justinmk** [comment]
This is the job of the _terminal_, it should be handling user input and then stopping the paste stream if necessary.

- The entire purpose of bracketed paste is for applications like Nvim to get a block of text for pasting, not processing. 
- Some terminals like macOS terminal.app have unforgivably slow bracketed paste. If they were faster, you wouldn't need to ctrl-c text unless it's hundreds of megabytes...

otoh, we are already inspecting each byte for the "stop paste" sigil. 

Supporting ctrl-c means that you can't paste text that contains a `^C` byte. That's probably a reasonable tradeoff?

This might be as easy as adding another variant here: https://github.com/justinmk/neovim/blob/307c5c63e17735aa493a43da6d514d00034e9b71/src/nvim/tui/input.c#L437

**Comment by timweri** [comment]
Hi @justinmk . I'm interested in implementing this.

I took a look at your proposal and tried to implement the simple fix. However, as you said, this is supposed to be the job of the terminal. I tested to see if the MacOS terminal would insert the CTRL-C halfway through the paste if I spam it. However, the large pasting didn't stop and the CTRL-C was registered after the pasting was complete.

Based on that I don't think interrupting a large paste with user input would be feasible, at least for the Terminal in question.
